src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -33,6 +38,23 @@ function useAuthState() {
     localStorage.getItem(NOTIFICATION_SERVICE_TOKEN_KEY)
   )
 
+  const handleAuthenticate = useCallback(() => {
+    setAuthState(AUTH_AUTHENTICATING)
+    isAuthTokenValid(token)
+      .then(v => {
+        setAuthState(AUTH_AUTHENTICATED)
+        return v
+      })
+      .catch(e => {
+        if (e instanceof TypeError) {
+          // network/service error
+          setAuthState(AUTH_SERVICE_UNAVAILABLE)
+        } else {
+          setAuthState(AUTH_AUTHENTICATION_FAILED)
```

Comment:
Shouldn't we check for an `UnauthorizedError` or `ExpiredTokenError`?

Or perhaps we should have a custom error for the service being unavailable (`ManageNotifcations` is also checking for `TypeError`)?

`TypeError` seems pretty generic to know that it the service is unavailable.

Reply:
> Shouldn't we check for an UnauthorizedError or ExpiredTokenError?

That depends if we want to handle the two cases the same way. In both cases we currently require the user the log in again, hence the general catch.

> TypeError seems pretty generic to know that it the service is unavailable.

`TypeError` is the standard error type for network errors. I didn't want to deviate too much from the fetch spec. https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Checking_that_the_fetch_was_successful

Reply:
Did not know that `TypeError` was the standard error type for `fetch()`!